src/store.ts

```diff
@@ -229,108 +236,139 @@ class Store extends ElectronStore<StoreData> {
     }
 
     /**
-     * Prepare the safeStorage backend for use.
-     * We don't eagerly import from keytar as that would bring in data for all Element profiles and not just the current one,
-     * so we import lazily in getSecret.
+     * Normalise the backend to a sane value (exclude `unknown`), respect forcePlaintext mode,
+     * and ensure that if an encrypted backend is picked that encryption is available, falling back to plaintext if not.
+     * @param forcePlaintext - whether to force plaintext mode
+     * @private
      */
-    private async prepareSafeStorage(): Promise<void> {
-        await app.whenReady();
+    private chooseBackend(forcePlaintext: boolean): SaneSafeStorageBackend {
+        if (forcePlaintext) {
+            return "plaintext";
+        }
 
-        let safeStorageBackend = this.get("safeStorageBackend");
+        const isEncryptionAvailable = safeStorage.isEncryptionAvailable();
         if (process.platform === "linux") {
+            // The following enables plain text encryption if the backend used is basic_text.
+            // It has no significance for any other backend.
+            // We do this early so that in case we end up using the basic_text backend (either because that's the only one available
+            // or as a fallback when the configured backend lacks encryption support), encryption is already turned on.
+            safeStorage.setUsePlainTextEncryption(true);
```

Comment:
This has to happen before computing `isEncryptionAvailable`. It's fine to call it first thing since it noops on windows/mac.